### DISCUSSION TITLE
Fully redundant cluster with 2 Datacenters

### DISCUSSION THREAD

**Discussion by strongshadow** [minio/minio#13952] [Q&A] [answered]
Hello! I would like to know your opinion regarding the following architecture.
We have 2 on-premises Datacenters with a high speed and low latency connection (10Gbit / s bandwidth / ~ 1ms latency). The intention is to build a distributed architecture of minio that has full redundancy, the possibility that either datacenter stops responding and the other one continues to operate in READ-WRITE mode (not READ-ONLY).

Since the Erasure Code set can tolerate a maximum of N / 2 - 1 nodes loss before going READ-ONLY, we have two architectures in mind.

The first one is to generate two different clusters of 4 nodes each (each cluster in a different DC) and maintain bidirectional replication at bucket level so that one cluster is a "mirror" of another.

The second one (and here comes the question) is to build an extended cluster between both DCs and generate a new node (+1) in a cloud provider (200Mbit / s bandwidth / ~ 150ms latency) to keep the amount of N / 2 + 1? This node would be only for the quorum and would not receive workloads at any time, only the synchronization with the other nodes.

Is it possible to build the second configuration? Did someone do it? What could be the problems involved?

Sorry if this discussion has already taken place, I couldn't find it.

Thanks a lot!

**Comment by klauspost** [comment, marked as answer]
Go the replication route, with at least 4 servers on each location. Latency alone makes the other setup infeasible.

**Reply by strongshadow** [reply]
@klauspost Thank you so much for your speedy reply!

**Reply by antverpp** [reply]
I wonder how do you manage to mitigate split brain scenaro in case of bidirectional replication ?

**Reply by klauspost** [reply]
@antverpp Versions are merged across async replicated clusters. So if cluster A has object X written while cluster B has another object X written both clusters will end up with 'object X' with 2 versions, ordered by modification time of each.
